lib/modules/manager/composer/extract.ts

```diff
@@ -127,9 +128,10 @@ export async function extractPackageFile(
               depType,
               depName,
               currentValue,
-              datasource: GithubTagsDatasource.id,
-              packageName: 'php/php-src',
-              extractVersion: '^php-(?<version>.*)$',
+              datasource: DockerDatasource.id,
+              extractVersion:
+                '^(?<version>\\d+\\.\\d+\\.\\d+(?:RC\\d+|beta\\d+|alpha\\d+)*)',
```

Comment:
```suggestion
                '^(?<version>\\d+\\.\\d+\\.\\d+(?:RC\\d+|beta\\d+|alpha\\d+)?)$',
```

- should only occure zero or one time, otherwise you match `1.2.3rc1beta5`
- without ending `$` you allow `1.2.3rc1-alpine`

Reply:
Good catch, it should be zero or one, not any, I will correct. As for the missing `$`, this is deliberate as I only want to capture what comes before the `-` and nothing after it.

Reply:
we only needs the bare versions, not the docker image specific ones. otherwise we later need to filter duplicates. so best is to filter them here.